internal/pkg/manifest/svc.go

```diff
@@ -224,6 +225,28 @@ func (a *AdvancedCount) unsetAutoscaling() {
 	a.Memory = nil
 	a.Requests = nil
 	a.ResponseTime = nil
+	a.QueueScaling = QueueScaling{}
+}
+
+// QueueScaling represents the configuration to scale a service based on a SQS queue.
+type QueueScaling struct {
+	AcceptableLatency *time.Duration `yaml:"acceptable_latency"`
+	AvgProcessingTime *time.Duration `yaml:"msg_processing_time"`
+}
+
+// IsEmpty returns true if the QueueScaling is set.
+func (qs *QueueScaling) IsEmpty() bool {
+	return qs.AcceptableLatency == nil && qs.AvgProcessingTime == nil
+}
+
+// AcceptableBacklogPerTask returns the total number of messages that each task can accumulate in the queue
+// while maintaining the AcceptableLatency given the AvgProcessingTime.
+func (qs *QueueScaling) AcceptableBacklogPerTask() (int, error) {
+	if err := qs.Validate(); err != nil {
```

Comment:
Maybe add a `TODO` so that we won' forget to remove it once we finish the manifest validation

Reply:
I wasn't sure if I should keep it or not even after we have manifest validation 🤔 maybe I can add a comment saying that this method should assume that the input is already Validated.

Reply:
ooo we should be careful to only call `AcceptableBacklogPerTask` when `qs` is not empty. Otherwise we will panic from proceeding to dereference `qs.AcceptableLatency` and `qs.AvgProcessingTime`

Reply:
yea 👍 that's why there is `Validate` here for now

Reply:
`Validate` returns `nil` error when `qs` is empty though! So it still seems possible to dereference a `qs. AcceptableLatency` that is `nil`, right? We should either make sure to call `AcceptableBacklogPerTask` only when appropriate or add an extra safeguard here, e.g. if it's empty, we return `0, errors.New("no queue autoscaling specified")`.